pulsar/internal/service_name_resolver.go

```diff
@@ -99,6 +104,8 @@ func (r *pulsarServiceNameResolver) UpdateServiceURL(u *url.URL) error {
 	r.ServiceURL = u
 	r.ServiceURI = uri
 	rand.Seed(time.Now().Unix()) // initialize global pseudo random generator
```

Comment:
This is kind of outside of the change but this should be done in the init function.